### PR TITLE
Implement Vultr cloud provider

### PR DESCRIPTION
**Pull request by jonatasbaldin**
**What this PR does / why we need it**:
This PR adds support to [Vultr](vultr.com/) cloud provider, for both their Bare Metal and Cloud Instance products. 

For now, is only supports `ubuntu`.

**What type of PR is this?**
/kind feature

**Special notes for your reviewer**:
- How does the automated E2E work? Who is responsible for the cloud provider token/billing?

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
- Add Vultr cloud provider
```

**Documentation**:

```documentation
- Included in the `docs/` folder.
```

### CODE REVIEW

**Comment by jonatasbaldin** [comment]
hey @ahmedwaleedmalik, thanks for moving on with the PR :)

I was wondering about the E2E tests, which require a Vultr API Token. How does that work? Who generates and/or is responsible for the token/bills?

**Comment by jonatasbaldin** [comment]
Fixed the lint issued.

As you can see on the failing licenses tests, the Vultr GO SDK depends on two projects by Hashicorp with MPL2 licenses. Is is ok to allow those dependencies?

Thanks!

**Comment by ahmedwaleedmalik** [comment]
@jonatasbaldin Thank you for the awesome contribution. It's always exciting to see the addition of new cloud providers to machine-controller.

Regarding the license, please add it to this ignore list https://github.com/kubermatic/machine-controller/blob/master/.wwhrd.yml#L28. MPL-2 is generally fine for us. 

> I was wondering about the E2E tests, which require a Vultr API Token. How does that work? Who generates and/or is responsible for the token/bills?

We at Kubermatic, unfortunately, don't have an account at Vultr. Making sure that this integration works, is usually a mandatory requirement for us. For most of the external PRs, we expect that the partner, customer, or individual contributor would provide us with a way to test these changes. Which in your case would be the Vultr API Token.

Would that be possible for you?

File: test/e2e/provisioning/all_e2e_test.go [hunk 1]
```diff
@@ -744,6 +746,42 @@ func TestHetznerProvisioningE2E(t *testing.T) {
 	runScenarios(t, selector, params, HZManifest, fmt.Sprintf("hz-%s", *testRunIdentifier))
 }
 
+// TestVultrBareMetalProvisioning - a test suite that exercises Vultr provider with a bare metal instance
+// by requesting nodes with different combination of container runtime type, container runtime version and the OS flavour.
+func TestVultrBareMetalProvisioningE2E(t *testing.T) {
```
Comment (moadqassem) [inline test/e2e/provisioning/all_e2e_test.go:751]: Those tests need to exist in prow as well: 
https://github.com/kubermatic/machine-controller/tree/master/.prow

However, for the time being, we don't have the infrastructural requirements to run the e2e tests for Vultr.

**Comment by moadqassem** [comment]
As @ahmedwaleedmalik already mentioned, for the time being, we cannot test this PR, however, we are working on it. Once we have the proper infra for Vultr we can proceed with this PR. Thanks a lot for your contribution :) 

/hold

**Comment by jonatasbaldin** [comment]
Hi there. We unfortunately can't provide a Vultr token for the E2E tests. We will get in touch with the Vultr team in search for a collaboration. I'll let you know if I have any news soon.

Thanks!

**Comment by ahmedwaleedmalik** [comment]
Closing this PR in favor of https://github.com/kubermatic/machine-controller/pull/1531. With https://github.com/kubermatic/machine-controller/pull/1531, support for Vultr is available in machine-controller 🎊

**Pull request closed without merging**